03.bowling/bowling_test.rb

```diff
@@ -0,0 +1,106 @@
+# frozen_string_literal: true
+
+require 'test/unit'
+require './bowling'
+
+class TestBowling < Test::Unit::TestCase
+  # split_by_framesのテスト
+  def testcase1_split_by_frames
```

Comment:
> テストケースが7つあるので愚直にナンバリングしている状態です😣

本来であれば連番はなるべく避けたいところですが、このテストパターンに名前を付けるのは難しいと思うので（よーく考えれば付けれそうですが）、この場合は例外的に連番でも良いと思います。

ちなみに僕だったらこんな感じの命名にすると思います。

```suggestion
  def test_split_by_frames_1
```

Reply:
コメントありがとうございます🙏
末尾に連番番号を持ってきただけなのに、読みやすさが違って見えますね...！👀

Reply:
テストコードに関して、いまさらのコメントになってしまいますが、理想的なテストコードは

`'6,3,9,0,0,3,8,2,7,3,X,9,1,8,0,X,6,4,5'`のような文字列を渡せば、139が返ってくるメソッドのテストを7パターン書く

ですね。
つまり、1つのテストでsplit_by_framesもcalculate_scoreも、両方同時にテストしてしまう方式です。

あと、test-unitだとデータ駆動テスト（parametrized test）も使えるので便利です。

```ruby
  data('test1', [139, '6,3,9,0,0,3,8,2,7,3,X,9,1,8,0,X,6,4,5'])
  data('test2', [164, '6,3,9,0,0,3,8,2,7,3,X,9,1,8,0,X,X,X,X'])
  def test_hogehoge(data)
    expected, scoreboard = data
    assert_equal(expected, hogehoge(scoreboard))
  end
```

https://www.clear-code.com/blog/2013/1/23.html

Reply:
上のコメントで書いたテスト、実際にやった方がいいですね。
テストを分けてしまったことで、不具合を検知できないテストになっているようです。
以下はJudeeeeeさんの最新のコードを僕のローカルで実行した結果です。

```
$ ruby bowling.rb 6,3,9,0,0,3,8,2,7,3,X,9,1,8,0,X,X,X,X
144
```

データ駆動にしなくてもいいですが、「'6,3,9,0,0,3,8,2,7,3,X,9,1,8,0,X,6,4,5'のような文字列を渡せば、139が返ってくるメソッドのテスト」は必須だと思いますー。